common/display/kmsfencehandler.cpp

```diff
@@ -82,6 +90,11 @@ void KMSFenceEventHandler::HandleRoutine() {
     close(kms_fence_);
     kms_fence_ = -1;
   }
+
+  for (NativeSurface *surface : surfaces_) {
+     surface->SetInUse(false);
```

Comment:
when we swap previous and current plane state, we are guaranteed that the previous plane state layers are not displayed anymore. We signal the overlaylayers in a separate thread to avoid case where previous layer are no more displayed but we cannot signal as app has not requested for next frame yet (thats we handle the wait fence in a separate thread and signal layers there). However, for nativesurface this shouldnt be an issue since app/surfaceflinger know nothing about native surface.